starter-code/src/QueueDataStructure.js

```diff
@@ -1,2 +1,34 @@
-function QueueDataStructure () {
+function QueueDataStructure() {
+  this.queueControl = [];
+  this.MAX_SIZE = 8;
 }
+
+QueueDataStructure.prototype.isEmpty = function() {
+  if (this.queueControl.length === 0) {
+    return true;
+  } else {
+    return false;
+  }
+};
+
+QueueDataStructure.prototype.canEnqueue = function() {
+  if (this.queueControl.length < this.MAX_SIZE) {
+    return true;
+  }
+};
+
+QueueDataStructure.prototype.enqueue = function(number) {
+  if (this.canEnqueue()) {
+    this.queueControl.push(number);
+    return this.queueControl;
+  } else {
+    return 'Queue Overflow';
+  }
+};
+
+
+QueueDataStructure.prototype.dequeue = function() {
+  if (this.isEmpty()) {
+    this.queueControl.unshift(this.queueControl.length[this.ququeueControl.length - 1]);
```

Comment:
Dequeue has to REMOVE the first element of the array. Unshift is used to ADD an element to the beginning of an array. Furthermore, I can't understand what are you trying to pass to unshift. Seriously, explain it to me later.

In conclusion, maybe you want to use shift.